src/Cardano/Wallet/Binary.hs

```diff
@@ -437,6 +440,45 @@ decodeUpdateProof = do
 
 -- * Encoding
 
+-- | Encode a public key to a corresponding Cardano Address. The encoding of the
+-- attributes part of an address is left out to the caller; This allows for
+-- distinguishing between Sequential and Random addresses (the former doesn't
+-- have any attributes to encode).
+--
+-- @
+-- -- Old / Random Addresses
+-- let encodeAttributes = mempty
+--      <> CBOR.encodeMapLen 1
+--      <> CBOR.encodeWord8 1
+--      <> encodeDerivationPath (hdPassphrase rootXPub) accIx addrIx
+-- let addr = encodeAddress xpub encodeAttributes
+--
+-- -- New / Sequential Addresses
+-- let encodeAttributes = mempty <> CBOR.encodeMapLen 0
+-- let addr = encodeAddress xpub encodeAttributes
+-- @
+encodeAddress :: XPub -> CBOR.Encoding -> CBOR.Encoding
```

Comment:
Are there any tests for this?

Reply:
Yes, we have golden tests comparing with Yoroi addresses.